File: src/address.ts
export interface Address {
    id: string,
    name: string,
    email: string,
    address: string,
    gender: Gender,
}

export enum Gender {
    Male = "Male",
    Female = "Female"
}

export function newAddress() : Address
{
    return {
        id: crypto.randomUUID(),
        name: "",
        email: "",
        address: "",
        gender: Gender.Female,
    }
}

File: src-tauri/src/address.rs
use serde::{Deserialize, Serialize};
use std::fmt::*;
use uuid::Uuid;

#[derive(Serialize, Deserialize, Clone)]
pub struct Address {
    pub id: Uuid,
    pub name: String,
    pub email: String,
    pub address: String,
    pub gender: Gender,
}

#[derive(Serialize, Deserialize, Clone, Debug)]
pub enum Gender {
    Male,
    Female,
}

impl Display for Address {
    fn fmt(&self, f: &mut Formatter<'_>) -> Result {
        write!(
            f,
            "{} | {} | {} | {:?}",
            self.id, self.name, self.address, self.gender
        )
    }
}
